lib/hydra/works/services/generic_work/move_to_generic_file.rb

```diff
@@ -0,0 +1,37 @@
+module Hydra::Works
+  class MoveGenericFileToGenericWork
+
+    ##
+    # Move a generic file from one generic work to another
+    #
+    # @param [Hydra::Works::GenericWork] :old_parent_work where the generic file currently lives
+    # @param [Hydra::Works::GenericWork] :new_parent_work where the generic file is being moved
+    # @param [Hydra::Works::GenericFile] :child_generic_file being moved
+    #
+    # @return [Hydra::Works::GenericWork] the destination generic work
+    def self.call(old_parent_work, new_parent_work, child_generic_file)
+      raise ArgumentError, "old_parent_work must be a Hydra::GenericWork" unless Hydra::Works.generic_work? old_parent_work
+      raise ArgumentError, "new_parent_work must be a Hydra::GenericWork" unless Hydra::Works.generic_work? new_parent_work
+      raise ArgumentError, "child_generic_file must be a Hydra::GenericFile" unless Hydra::Works.generic_file? child_generic_file
+      add_file_to_work new_parent_work, child_generic_file
+      remove_file_from_work old_parent_work, child_generic_file
+      new_parent_work.reload
+    end
+
+    private
+
+    def self.add_file_to_work(work, file) 
+      # TODO: This should be changed to work.generic_files once 
+      #       generic_files has been fully implemented in Hydra::Works
+      work.members << file
```

Comment:
@no-reply I like your idea which coincidentally matches the approach that @elrayle was suggesting during the sprint call earlier today.